packages/plugin-react-refresh/index.js

```diff
@@ -82,6 +84,9 @@ function reactRefreshPlugin(opts) {
           plugins: opts && opts.parserPlugins
         },
         plugins: [
```

Comment:
For `syntax-jsx` and TS handling: `@babel/parser` actually ships plugins for these directly - we just need to enable them in `parserOpts.plugins` above. See https://babeljs.io/docs/en/babel-parser#plugins

So:

- Enable `jsx` by default, which can get rid of the need for `@babel/plugin-syntax-jsx`
- Enable `typescript` in plugins if file extension is TS/TSX

Reply:
Done.